agent/api/container/container.go

```diff
@@ -910,3 +945,56 @@ func (c *Container) GetStopTimeout() time.Duration {
 
 	return time.Duration(c.StopTimeout) * time.Second
 }
+
+func (c *Container) SetRestartAttempts(count RestartCount) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	c.RestartAttempts = count
+}
+
+func (c *Container) GetRestartAttempts() RestartCount {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+	return c.RestartAttempts
+}
+
+func (c *Container) IncrementRestartAttempts() {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	c.RestartAttempts += 1
+}
+
+func (c *Container) CanMakeRestartAttempt() bool {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	return c.RestartPolicy == Always ||
+		c.RestartPolicy == OnFailure &&
+		c.RestartAttempts < c.RestartMaxAttempts
+}
+
+func (c *Container) IsAutoRestartNonEssentialContainer() bool {
+	return !c.IsEssential() && c.RestartPolicy != NEVER
+}
+
+func (c *Container) IsDesiredToRestartWhenReceivingStopped() bool {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	defer func() {
```

Comment:
This is used for checking if `DesiredToRestartWhenReceivingStopped` is true, and if it's true set it as false. May be I can remove the lock, since we are not calling `handleContainerChange()` asynchronously and the flag is only set in this function?